client/src/JsAgent/JsAgent.js

```diff
@@ -0,0 +1,63 @@
+const axios = require('axios');
+const Channel = require('jschannel');
+const URI = require('urijs');
+
+const instance = axios.create({
+  validateStatus: () => true
+});
+
+const Apiary = window.Apiary || {};
+Apiary.channels = [];
+
+//This function is exposed for testing purposes.
+const buildChannel = Apiary.buildChannel = function buildChannel(origin, scope) {
+  const chan = Channel.build({
+    window: window.parent,
+    origin: origin,
+    scope: scope
+  });
+
+  chan.bind('httpRequest', (trans, requestOptions) => {
+    trans.delayReturn(true);
+
+    instance(requestOptions)
+      .then(trans.complete)
+      .catch((err) => {
+        let errorCopy = JSON.parse(JSON.stringify(err));
+        errorCopy.message = err.message;
+        trans.error(errorCopy);
+      })
+  });
+
+  return chan;
+};
+
+Apiary.createAgent = function createAgent({ subdomains }) {
```

Comment:
Is there a reason why are you using an `object` with a single property `subdomains` property instead of `array`?